### ISSUE TITLE
Training stuck on 'INFO:torch.nn.parallel.distributed:Reducer buckets have been rebuilt in this iteration.'

### ISSUE THREAD

**Issue by farcast3r** [RVC-Project/Retrieval-based-Voice-Conversion-WebUI#1820]
I'm trying to resume training from around 6 months ago and I'm getting a few errors including some about parameters being deprecated, and then the line _INFO:torch.nn.parallel.distributed:Reducer buckets have been rebuilt in this iteration._ appears and then nothing seems to happen.

I left it running for a while yesterday, and even though when training previously, epochs took a considerable amount of time to build, I was surprised to not have anything show up in the console for 8+ hours. The cmd.exe was still using 15%~ CPU so it hadn't completely frozen, but I just wondered if someone could check my logs please in case my files are corrupted (or too old to continue training) and I would be better off starting from scratch? Or is this normal and I just need to wait longer?

`
C:\Users\Jake\Downloads\RVC-beta\RVC-beta0717>runtime\python.exe infer-web.py --pycmd runtime\python.exe --port 7897
Found GPU NVIDIA GeForce GT 1030
Use Language: en_US
Running on local URL:  http://0.0.0.0:7897
write filelist done
use gpus: 0
runtime\python.exe train_nsf_sim_cache_sid_load_pretrain.py -e ash -sr 40k -f0 1 -bs 13 -g 0 -te 200 -se 10 -pg pretrained_v2/f0G40k.pth -pd pretrained_v2/f0D40k.pth -l 0 -c 0 -sw 1 -v v2
INFO:ash:{'train': {'log_interval': 200, 'seed': 1234, 'epochs': 20000, 'learning_rate': 0.0001, 'betas': [0.8, 0.99], 'eps': 1e-09, 'batch_size': 13, 'fp16_run': False, 'lr_decay': 0.999875, 'segment_size': 12800, 'init_lr_ratio': 1, 'warmup_epochs': 0, 'c_mel': 45, 'c_kl': 1.0}, 'data': {'max_wav_value': 32768.0, 'sampling_rate': 40000, 'filter_length': 2048, 'hop_length': 400, 'win_length': 2048, 'n_mel_channels': 125, 'mel_fmin': 0.0, 'mel_fmax': None, 'training_files': './logs\\ash/filelist.txt'}, 'model': {'inter_channels': 192, 'hidden_channels': 192, 'filter_channels': 768, 'n_heads': 2, 'n_layers': 6, 'kernel_size': 3, 'p_dropout': 0, 'resblock': '1', 'resblock_kernel_sizes': [3, 7, 11], 'resblock_dilation_sizes': [[1, 3, 5], [1, 3, 5], [1, 3, 5]], 'upsample_rates': [10, 10, 2, 2], 'upsample_initial_channel': 512, 'upsample_kernel_sizes': [16, 16, 4, 4], 'use_spectral_norm': False, 'gin_channels': 256, 'spk_embed_dim': 109}, 'model_dir': './logs\\ash', 'experiment_dir': './logs\\ash', 'save_every_epoch': 10, 'name': 'ash', 'total_epoch': 200, 'pretrainG': 'pretrained_v2/f0G40k.pth', 'pretrainD': 'pretrained_v2/f0D40k.pth', 'version': 'v2', 'gpus': '0', 'sample_rate': '40k', 'if_f0': 1, 'if_latest': 0, 'save_every_weights': '1', 'if_cache_data_in_gpu': 0}
INFO:torch.distributed.distributed_c10d:Added key: store_based_barrier_key:1 to store for rank: 0
INFO:torch.distributed.distributed_c10d:Rank 0: Completed store-based barrier for key:store_based_barrier_key:1 with 1 nodes.
gin_channels: 256 self.spk_embed_dim: 109
./logs\ash\D_2333333.pth
INFO:ash:Loaded model weights
INFO:ash:Loaded checkpoint './logs\ash\D_2333333.pth' (epoch 30)
INFO:ash:loaded D
./logs\ash\G_2333333.pth
INFO:ash:Loaded model weights
INFO:ash:Loaded checkpoint './logs\ash\G_2333333.pth' (epoch 30)
C:\Users\Jake\Downloads\RVC-beta\RVC-beta0717\runtime\lib\site-packages\torch\functional.py:641: UserWarning: stft with return_complex=False is deprecated. In a future pytorch release, stft will return complex tensors for all inputs, and return_complex=False will raise an error.
Note: you can still call torch.view_as_real on the complex output to recover the old return format. (Triggered internally at C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\aten\src\ATen\native\SpectralOps.cpp:867.)
  return _VF.stft(input, n_fft, hop_length, win_length, window,  # type: ignore[attr-defined]
INFO:torch.nn.parallel.distributed:Reducer buckets have been rebuilt in this iteration.
C:\Users\Jake\Downloads\RVC-beta\RVC-beta0717\runtime\lib\site-packages\torch\autograd\__init__.py:200: UserWarning: Grad strides do not match bucket view strides. This may indicate grad was not created according to the gradient layout contract, or that the param's strides changed since DDP was constructed.  This is not an error, but may impair performance.
grad.sizes() = [64, 1, 4], strides() = [4, 1, 1]
bucket_view.sizes() = [64, 1, 4], strides() = [4, 4, 1] (Triggered internally at C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\torch\csrc\distributed\c10d\reducer.cpp:337.)
  Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
INFO:torch.nn.parallel.distributed:Reducer buckets have been rebuilt in this iteration.
`

[train.log](https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/files/14309958/train.log)

**Comment by farcast3r** [comment]
Sorry please ignore this, for anyone else wondering it just takes a little extra time for the checkpoint epoch to generate and then subsequent training to continue, this was probably evident from the fact that the cmd.exe was still using a lot of CPU.

Just my two cents but I think it would be helpful and a bit more user-friendly if the logger printed a message to show that training had resumed successfully.
